### DISCUSSION TITLE
Unrecognized function or variable 'y_s'

### DISCUSSION THREAD

**Discussion by ArjunM21** [yalmip/YALMIP#1069] [Problems using YALMIP] [answered]
Hello Johan.

Until recently I had been solving problems using YALMIP comfortably, Recently when I tried to run a large problem, suddenly I got an error saying "Nope, unexplained crash in SeDuMi! (could be memory issues or wrong binary) Make sure you have a recent and compiled version". I tried debugging it and the error said "Undefined function or variable 'y_s'
error callsedumi (line 75)
Primal = y_s;"

I am not sure what happened, but I tried reinstalling YALMIP again, but again the same error has popped up. Can you help me understand what has happened and give me a solution to this? I urgently need to fix this problem. Please find attached the file I use to install this
[YALMIP.zip](https://github.com/yalmip/YALMIP/files/7874768/YALMIP.zip)
.

**Comment by johanlofberg** [comment, marked as answer]
If SeDuMi works on other problems, then the error message is clearly indicating that you have a memory issue, which is consistent with solving a larger problem

**Reply by ArjunM21** [reply]
Yeah, I understand, but sedumi isn't working for any of the problems now. Even the small problems. Also, I checked my code on another laptop and it worked fine and ran without error. So I am guessing something happened to my yalmip toolbox?

**Reply by johanlofberg** [reply]
No, then your sedumi installation is broken, i.e. already sedumi(1,1,1) crashes

**Reply by ArjunM21** [reply]
Yeah, sedumi(1,1,1) gives an error. What I don't understand is how come a random day this has happened, when it used to work fine until the previous day? I have not changed anything except the problem size, but now even the smaller ones don't work. Do you have any suggestions on how to fix the issue? Should I try installing sedumi again?

**Reply by johanlofberg** [reply]
the error is telling you what is wrong and thus what to do

**Reply by ArjunM21** [reply]
Okay. I am attaching the error message that I get when I run the program. Can you help me understand it better?
[Error.pdf](https://github.com/yalmip/YALMIP/files/7888724/Error.pdf)

**Reply by johanlofberg** [reply]
Hard to say as it is somewhat silenced by the fact that it is called inside the MISDP solver. You will have to put a break in callsedumi where the call is made and manually run that line, to see what happens when it crashes. It might also be revealed if you crank up verbose to 2

**Reply by ArjunM21** [reply]
Okay, I tried setting verbose to 2, but the same error was shown. Putting a break in callsedumi means I have to write it in sedumi.m file? Also, how do I manually run that line after?

**Reply by johanlofberg** [reply]
You just put a break on the line where the call to sedumi is made, and then when you arrive there you run that line in the command line to see what happens, or any other strategy you might want to use to make sure you understand what happens when sedumi crashes
https://se.mathworks.com/help/matlab/matlab_prog/set-breakpoints.html

Or you set relax to 1 and solver to sedumi, and most likely it will crash on the continuous relaxation too

**Reply by ArjunM21** [reply]
Thank you for your help.
So I added a break line to the problem but when I run the next line in the command window, I get the exact same error, " 
Unrecognized function or variable 'y_s'". Should I try running this with another solver? Also pls explain "Or you set relax to 1 and solver to sedumi, and most likely it will crash on the continuous relaxation too" to me.

**Reply by johanlofberg** [reply]
You should put a break on the line " [x_s,y_s,info] = sedumi(-F_struc(:,2:end),-c,F_struc(:,1),K,pars);" and when the code stops there, you copy "sedumi(-F_struc(:,2:end),-c,F_struc(:,1),K,pars);" and run that in the command window, i.e. you manually make the call and observe what happens

**Reply by johanlofberg** [reply]
But above you said sedumi simply didn't work. Have you sorted that out? I hope you're not wasting time with a complex MISDP and trying to debug seumi, if you haven't fixed the obviously flawed sedumi installation to begin with

**Reply by ArjunM21** [reply]
Hey, thank you.
Sedumi is running but it crashes immediately. It crashes even for small problems. I reinstalled everything properly but still, it's not wokring properly. 
Using your advice, I ran the code with breaks and saw that an error comes from the callsedumi.m file as seen below. Can you explain what's wrong now? 
[Error 2.pdf](https://github.com/yalmip/YALMIP/files/7895984/Error.2.pdf)

**Reply by johanlofberg** [reply]
So why are you reporting what happens when you use sedumi via yalmip via bnb, instead of simply reporting what happens when you call sedumi directly...

Anyway, it obviously shows that there is a function called symbchol which doesn't like that sedumi sends as an argument to it.

Your installation of sedumi is broken somehow. I have no idea what you've done or what version you are using as sedumi as of recent years does not make a call to symbchol on line 341, nor are there any calls to symbchol with an argument
https://github.com/sqlp/sedumi/blob/c24a9e383c52d2f0c93e7f792068ba9479af7797/sedumi.m#L341

**Reply by ArjunM21** [reply]
Thank you so much for your help.
How do you want me to call sedumi directly? When I type in the command window what you asked me to, the error shown in the above pdf shows up. Can you pls explain how you want me to run sedumi directly? 
Also, I understand the installation might be broken now, but then how come it worked fine for all this time until the first crash?
Also, I tried running using the sedumi code that you sent above, but got the same error "Undefined function or variable 'y_s'
error callsedumi (line 75)
Primal = y_s;"
I understand this error is coming up because the sedumi.m file is not running, but even though I replaced the code of sedumi.m with the code you sent, I am getting the same error. I am unsure what to do now. 
I saw the same kind of question that has been asked before {https://github.com/sqlp/sedumi/blob/c24a9e383c52d2f0c93e7f792068ba9479af7797/sedumi.m#L341} but the author found some solution which he hadn't explained. 

I know we are going in circles, but I don't know what else to do other than ask here and try and figure the problem out.

**Reply by johanlofberg** [reply]
You run sedumi(1,1,1) to see what happens. As I said, your current installation of sedumi is broken, as the error message shown when sedumi ran indicates an error on a line that simply cannot happen on that line in the current version

**Reply by johanlofberg** [reply]
What have you replaced? You should not edit any sedumi file (or any file for that matter)

**Reply by ArjunM21** [reply]
When I run sedumi(1,1,1) I get the following error:
Error using pretransfo (line 163)
Should have length(c) > length(b) in any sensible model

Error in sedumi (line 253)
[A,b,c,K,prep,origcoeff] = pretransfo(A,b,c,K,pars);

Do you think I should install sedumi again? I have sedumi 1.3 currently. And I use this file to install all of this and yalmip called install_mpt3.m which I downloaded from github. I included that file in the first query I asked.

**Reply by johanlofberg** [reply]
You are using a very old version of sedumi.

**Reply by johanlofberg** [reply]
The error you showed can only occur in versions older than 2013
https://github.com/sqlp/sedumi/commit/5abacde0e63a50d5103d9a906125c045beb9f82f#diff-149f8091aaf30693d3ed066216ab64b7afda3a23869030a4bd92f0e16042025b

**Reply by ArjunM21** [reply]
Okay got it. So where can I find the latest version? Also, can you help me in adding paths to it later on?

**Reply by johanlofberg** [reply]
On github, which all links above have been pointing to.

**Reply by ArjunM21** [reply]
Hey, thank you for all your help.

So I removed all solvers from MATLAB that I had installed and installed YALMIP separately and also tried running yalmiptest with success. I ran a small program and it ran with the help of LMILAB. After this, I installed the latest sedumi package and when I tried to run sedumi(1,1,1), it gave an error saying 
"Unrecognized field name "cpx".

Error in sedumi (line 254)
if prep.cpx.dim>0"

Can you advise me on what I should try next? Seems like yalmip toolbox works, but I need to install sedumi somehow without error.

**Reply by johanlofberg** [reply]
I have no idea how you have managed to install something else than the official latest version which we talk about here, which clearly cannot raise that error
https://github.com/sqlp/sedumi/blob/c24a9e383c52d2f0c93e7f792068ba9479af7797/sedumi.m#L254

**Reply by ArjunM21** [reply]
I did download it from here. I downloaded a zip file called sedumi-1.3.5.zip, unzipped it, and then added the path to it. After this, I ran sedumi(1,1,1), which is when I got that error mentioned above. If it is not too much trouble, can you give me step-by-step instructions on how to proceed so that I can make sure I do nothing wrong?

**Reply by johanlofberg** [reply]
then you have a messed up installation with multiple copies, because as I said, master branch, which 1.3.5 is tagged on, does not do what is printed on line 254. Here explicitly shown on the 1.3.5-tagged hash
https://github.com/sqlp/sedumi/blob/671741c4ae8d216e696c2c56441bdc54fd191fb6/sedumi.m#L254

**Reply by johanlofberg** [reply]
and simply googling if prep.cpx.dim>0 to see where that can be found gives a single link to some old sedumi version inside some 3rd-party toolbox on matlab central

**Reply by johanlofberg** [reply]
i.e. which sedumi -all will reveal multiple copies on your path

**Comment by ArjunM21** [comment]
Thank you so much.
I have figured it out now. Everything runs properly, although for a specific program given below, it is showing an error. But anyway, thank you so much for everything.

```
clc;
clear all;
A = [1 0 0 ; 0 1 0; 0 1 0; 0 1 0; 0 0 1; 0 1 -1;-1 1 0; 1 -1 1]; %1,2,5
q = binvar(8,1);
Q = diag(q);
Y = sdpvar(8);
M = 3;
Model = [[Y A;A' A'*Q*A] >= 0,sum(q)==M];
% options = sdpsettings('solver','bnb');
optimize(Model,trace(Y))
```

**Reply by johanlofberg** [reply]
solves fine here

```
   47 :    1.100E+01     0.46      1.090E+01   1       3.6    Successfully solved  
   48 :    1.100E+01     0.46      1.090E+01   0       3.6    Successfully solved  
   49 :    1.100E+01     0.00      1.090E+01   0       3.7    Successfully solved  
+  49 Finishing.  Cost: 11

ans = 

  struct with fields:

    yalmipversion: '20210331'
    matlabversion: '9.9.0.1524771 (R2020b) Update 2'
       yalmiptime: 1
       solvertime: 3.7650
             info: 'Successfully solved (BNB)'
          problem: 0
```

**Reply by ArjunM21** [reply]
Hey just another question. Is there a way to adjust the max time taken for solving a problem using sedumi and also the gap% value to be considered before settling on a solution? I do not see these in the sedumi parameter list.

**Reply by johanlofberg** [reply]
Well, then you cannot.
